app/src/shared/web3/web3Poller.js

```diff
@@ -0,0 +1,192 @@
+// @flow
+
+import EventEmitter from 'events'
+import { I18n } from 'react-redux-i18n'
+
+import getWeb3, { getPublicWeb3 } from '$shared/web3/web3Provider'
+import type { StreamrWeb3 as StreamrWeb3Type } from '$shared/web3/web3Provider'
+import { areAddressesEqual } from '$mp/utils/smartContract'
+import type { NumberString } from '$shared/flowtype/common-types'
+import { hasTransactionCompleted } from '$shared/utils/web3'
+import { getTransactionsFromSessionStorage } from '$mp/modules/transactions/services'
+import TransactionError from '$shared/errors/TransactionError'
+
+const events = {
+    ACCOUNT: 'WEB3POLLER/ACCOUNT',
+    ACCOUNT_ERROR: 'WEB3POLLER/ACCOUNT_ERROR',
+    NETWORK: 'WEB3POLLER/NETWORK',
+    NETWORK_ERROR: 'WEB3POLLER/NETWORK_ERROR',
+    TRANSACTION_COMPLETE: 'WEB3POLLER/TRANSACTION_COMPLETE',
+    TRANSACTION_ERROR: 'WEB3POLLER/TRANSACTION_ERROR',
+}
+
+type Event = $Values<typeof events>
+type Handler = (any, any) => void
+
+const ONE_SECOND = 1000
+const FIVE_SECONDS = 1000 * 5
+
+class Web3Poller {
+    web3PollTimeout: ?TimeoutID = null
+    ethereumNetworkPollTimeout: ?TimeoutID = null
+    pendingTransactionsPollTimeout: ?TimeoutID = null
+    web3: StreamrWeb3Type = getWeb3()
+    account: any = null
+    networkId: NumberString = ''
+    emitter: EventEmitter = new EventEmitter()
+
+    constructor() {
+        // Start polling for info
+        this.pollWeb3()
+        this.pollEthereumNetwork()
+        this.pollPendingTransactions()
+    }
+
+    subscribe(event: Event, handler: Handler) {
+        this.emitter.on(event, handler)
+    }
+
+    unsubscribe(event: Event, handler: Handler) {
+        this.emitter.removeListener(event, handler)
+    }
+
+    pollWeb3 = () => {
+        this.fetchWeb3Account()
+        this.clearWeb3Poll()
+        this.web3PollTimeout = setTimeout(this.pollWeb3, ONE_SECOND)
```

Comment:
What happens if `fetchWeb3Account` takes longer than one second for whatever reason? it looks like it'll continue to enqueue more `fetchWeb3Account` calls every second without waiting for the previous one to resolve. If you return a promise from `fetchWeb3Account` then you could chain the `setTimeout` to occur only after `fetchWeb3Account` finishes, and perhaps should add a check to ensure it's not already running before starting a new one.

Reply:
Good point, changed it so that the new timeout is set after the promise completes. The old one is cleared in eg. `this.clearWeb3Poll` so basically we shouldn't end up in a situation where a timer is already set?